avocado/utils/process.py

```diff
@@ -750,7 +756,13 @@ def send_signal(self, sig):
         :param sig: Signal to send.
         """
         self._init_subprocess()
-        self._popen.send_signal(sig)
+        if self.is_running_as_sudo():
+            for child in get_children_pids(self.get_pid()):
+                child_pid = child.decode('UTF-8')
+                kill_child_cmd = 'kill -%d %s' % (sig.value, child_pid)
+                run(kill_child_cmd, sudo=True)
```

Comment:
Maybe we need to treat (ignore some) errors here because there may be race conditions where some of the processes that we got on `get_children_pid()` may not exist anymore.